copilot-chat.el

```diff
@@ -280,11 +279,27 @@ Argument PROMPT is the prompt to send to Copilot."
 (defun copilot-chat--insert-and-send-prompt (prompt)
   "Helper function to prepare buffers and send PROMPT to Copilot.
 This function may be overriden by frontend."
-  (copilot-chat--prepare-buffers)
-  (with-current-buffer copilot-chat--prompt-buffer
-    (erase-buffer)
-    (insert prompt))
-  (copilot-chat-prompt-send))
+  (let* ((prompt-suffix (copilot-chat--build-prompt-suffix))
+         (final-prompt (if prompt-suffix
+                           (concat prompt "\n" prompt-suffix)
+                         prompt)))
+    (copilot-chat--prepare-buffers)
+    (with-current-buffer copilot-chat--prompt-buffer
+      (erase-buffer)
+      (insert final-prompt))
+    (copilot-chat-prompt-send)))
+
+(defun copilot-chat--build-prompt-suffix ()
+    "Build a prompt suffix with the current buffer name."
+    (if (derived-mode-p 'prog-mode)  ; current buffer is a programming language buffer
+        (let* ((major-mode-str (symbol-name major-mode))
+               (lang (replace-regexp-in-string "-mode$" "" major-mode-str))
+               (dynamic-suffix (format "current programming language is: %s" lang))
+               (suffix (if copilot-chat-prompt-suffix
```

Comment:
I think the static suffix should be added in `copilot-chat--create-req` to be added to all prompts, not only when using "shortcut" functions.
(The suffix could be a great way to be sure copilot will use org syntax when using org backend)

Reply:
I see. Agree that only static part might be useful for other place. Cause the dynamic-suffix contains buffer major mode specific information. The copilot-chat--create-req was invoked by copilot-chat--ask, and the later function are used by

1. copilot-chat--insert-and-send-prompt (shortcut mostly use this)
2. copilot-chat-ask-and-insert (dedicate interactive function. not invoked by other functions)
3. copilot-chat-insert-commit-message

Do we want to apply the static suffix to all three entries above?

Reply:
I think this could be a good thing. What do you think ?

Reply:
The initial purpose to have static suffix, is to make it easier to read the output from copilot-chat. There usually a lot. It is very useful to me since English is not my first language. And to my own using experience, the discussion with copilot-chat take most of time, so, make it easier read to me is very helpful.
copilot-chat-ask-and-insert is about to send to Copilot a custom prompt and insert answer in current buffer at point. It might make sense to have static suffix attached but it also depends, cause it directly write output to the current buffer. (eg. Does the translation really required?)
For copilot-chat-insert-commit-message, I am not sure if the static suffix will be useful here. This is a pretty special use case.

Reply:
@chep Let me know if it make sense. Right now I feel that the code / copilot-chat view is the most useful to me (copilot-chat--insert-and-send-prompt), and it works well for the suffix. I can add it to others (personally feel copilot-chat-ask-and-insert is relative relevant, but it is less likely for me to let copilot-chat write to my current working buffer, since not all the output is useful for the code). Or we can add to them when needed.

Reply:
Hi, sorry I forgot to reply. Let's merge it and I will see later if the static suffix can be useful elsewhere.
Thank you again.